examples/plugins/src/file-size.plugin.unit.test.ts

```diff
@@ -0,0 +1,361 @@
+import { formatBytes } from '@code-pushup/utils';
+import {
+  assertFileSize,
+  errorMessage,
+  fileSizeIssues,
+  infoMessage,
+  runnerFunction,
+  scoreFilesizeAudit,
+} from './file-size.plugin';
+
+// Mock file system API's
+vi.mock('fs', async () => {
+  const memfs: typeof import('memfs') = await vi.importActual('memfs');
+  return memfs.fs;
+});
+vi.mock('fs/promises', async () => {
+  const memfs: typeof import('memfs') = await vi.importActual('memfs');
+  return memfs.fs.promises;
+});
+
+const outputDir = MEMFS_VOLUME;
+const projectJson = JSON.stringify(
+  {
+    test: 42,
+    arr: [1, 2, 3],
+    obj: { test: 42 },
+  },
+  null,
+  2,
+);
+const testJs = `
+    const str = 'Hello World'
+    const num = 42;
+    const obj = ${projectJson};
+  `;
+const readmeMd = '# Docs';
+
+const testFile = 'test.js';
+
+describe('infoMessage', () => {
+  it.each([['index.js'], [join('src', 'index.js')]])(
+    'should return info message',
+    file => {
+      expect(infoMessage(file, 12)).toBe(
+        `File ${basename(file)} is OK. (size: ${formatBytes(12)})`,
+      );
+    },
+  );
+});
+
+describe('errorMessage', () => {
+  it.each([
+    [1, 0],
+    [2, 1],
+  ])(
+    'should return error message for size %i with budget %i',
+    (size, budget) => {
+      expect(errorMessage('test.js', size, budget)).toBe(
+        `File ${testFile} has ${size} B, this is ${1} B too big. (budget: ${budget} B)`,
+      );
+    },
+  );
+});
+
+describe('assertFileSize', () => {
+  it.each([[-1], [0], [1]])(
+    'should return a informative Issue without budgets (size: %s)',
+    size => {
+      expect(assertFileSize(testFile, size)).toEqual({
+        message: infoMessage(testFile, size),
+        severity: 'info',
+        source: { file: testFile },
+      });
+    },
+  );
+
+  it.each([
+    [0, 0],
+    [0, 1],
+    [1, 1],
+  ])(
+    'should return a informative Issue with budgets not exceeding (size: %s, budget: %s)',
+    (size, budget) => {
+      expect(assertFileSize(testFile, size, budget)).toEqual({
+        message: infoMessage(testFile, size),
+        severity: 'info',
+        source: { file: testFile },
+      });
+    },
+  );
+
+  it.each([[1, 0]])(
+    'should return error Issue with budgets exceeding (size: %s, budget: %s)',
+    (size, budget) => {
+      expect(assertFileSize(testFile, size, budget)).toEqual({
+        message: errorMessage(testFile, size, budget),
+        severity: 'error',
+        source: { file: testFile },
+      });
+    },
+  );
+});
+
+describe('scoreFilesizeAudit', () => {
+  it.each([
+    [-1, -1, 1],
+    [0, -1, 1],
+    [0, 0, 1],
+    [1, 0, 1],
+    [2, 1, 0.5],
+    [2, 2, 0],
+  ])(
+    'should return correct score (files: %s, errors: %s, score: %s)',
+    (files, errors, score) => {
+      expect(scoreFilesizeAudit(files, errors)).toBe(score);
+    },
+  );
+});
+
+describe('fileSizePlugin', () => {
+  beforeEach(() => {
+    vol.reset();
```

Comment:
Similarly to the fs mocks, you can also include `reset.mocks.ts` from the `testing-utils` `setup` folder that automatically resets the volume before each test.